### ISSUE TITLE
PushT evaluation reports 2% success while policy appears not to move the environment

### ISSUE THREAD

**Issue by sige0002** [sige0002/bt-sigreg#20]
## Problem

A PushT evaluation completes over 50 unique episodes but reports only 2% success, and the observed rollout appears to barely move / not move meaningfully at all.

Observed result:

```text
evaluated_unique_episodes: 50
start_steps: [14, 51, 6, 40, 32, 24, 43, 70, 51, 85, 22, 80, 48, 15, 21, 8, 56, 44, 20, 65, 40, 44, 25, 36, 20, 15, 1, 75, 1, 72, 14, 39, 117, 16, 22, 24, 92, 80, 55, 50, 53, 27, 45, 63, 76, 82, 61, 2, 55, 35]
{'success_rate': 2.0,
 'episode_successes': array([False, ..., True, ..., False]),
 'seeds': None}
```

The 2% value is numerically consistent with 1 success out of 50 episodes, but the more concerning symptom is that the evaluated policy appears effectively stationary. This makes it unclear whether the result reflects model quality or an evaluation/action-path bug.

## Relevant evaluation path

`lewm/eval.py` loads checkpoint-specific action statistics and, when `eval.shared_physical_search=true`, wraps the model with `PlanningActionAdapter`:

```python
model = PlanningActionAdapter(
    model,
    process['action'].mean_,
    process['action'].scale_,
    mean,
    std,
).eval()
```

`PlanningActionAdapter.normalize_for_model()` currently performs:

```python
converted = (
    shaped * self.reference_std
    + self.reference_mean
    - self.training_mean
) / self.training_std
```

The CEM search therefore operates in one normalized action space while the model receives checkpoint-normalized actions. A mismatch in the normalization contract, scale semantics, or action adapter direction could make model-scored actions inconsistent with the physical actions sent to the PushT environment.

## Need to distinguish evaluation bug from undertrained checkpoint

Before interpreting low success as model performance, verify the complete action path:

1. CEM candidate distribution before model normalization.
2. Candidate actions after `PlanningActionAdapter.normalize_for_model()`.
3. Selected CEM action / action chunk.
4. Exact physical action passed to `world.envs.step()`.
5. Resulting state displacement in the environment.

The evaluator already contains an audit hook that records `physical_actions`; extend/use this so the run emits useful summary diagnostics rather than requiring manual inspection of a large raw structure.

## Proposed diagnostics

For `--cem-audit`, record at minimum per run and preferably per case:

- physical action min/max/mean/std
- mean absolute action magnitude
- fraction of actions near zero (with an explicit threshold)
- CEM candidate min/max/mean/std before and after adapter normalization
- selected-plan min/max/mean/std
- initial and final agent/object state displacement
- number of actual environment step calls
- active-case mask evolution

Also print the action normalization statistics used:

```text
reference_mean
reference_std
training_mean
training_std
```

and verify that their meaning is consistent (`StandardScaler.scale_` vs true std, dimensionality, action units, and ordering).

## Control experiments

Add a minimal diagnostic comparison using the same manifest/cases:

1. random policy
2. a known-good/reference LeWM checkpoint
3. the current BT-SIGReg checkpoint

If the reference checkpoint also produces effectively stationary rollouts, treat this as an evaluation-stack bug. If only the BT checkpoint does so while action plumbing is correct, then the low success is likely due to the checkpoint/model itself.

## Acceptance criteria

- `--cem-audit` provides enough information to verify that nontrivial physical actions are actually sent to the environment.
- Action normalization before model scoring is explicitly checked against the checkpoint preprocessing contract.
- The evaluator can distinguish `policy produced near-zero actions`, `CEM selected near-zero actions`, and `environment did not respond to non-zero actions`.
- A known-good/reference checkpoint is tested through the exact same evaluation path.
- Low success is not reported as a trustworthy model-performance conclusion when action-path diagnostics indicate an evaluation failure.
- Add a regression/smoke test that asserts non-zero commanded actions lead to measurable PushT state displacement under the evaluation environment.

**Comment by sige0002** [comment]
調査で2点を確認しました。明示HDF5パスのローダーが keys_to_cache を keys_to_load として渡し、pixels 列を除外していました。結果として指定ケースの開始画像・Goal画像が評価へ渡りません。また SWM EnvPool は画像配列をインプレース更新しますが、動画収集側はそのビューを保持するため過去フレームも変わります。ローダー修正とstepごとの独立スナップショット、候補正規化前後・物理行動・実状態移動の診断を追加中です。実PushT 3環境で非ゼロ行動による移動とフレーム保持を回帰確認（関連7件合格）、CPU全体117件合格・CUDA専用5件未実行。現在、公式checkpointを固定confirm先頭50ケース・標準CEM予算で実評価中です。1ケースの完走を性能検証とは扱いません。対照評価の結果が揃うまで本Issueは閉じません。

**Comment by sige0002** [comment]
追加の実測です。修正途中の50ケース監査でメモリ急増を検出し、その評価のみ中断しました（failed、ログ保持）。既存CEM監査が get_cost の変更後の展開画像テンソルを各反復で保持していたためです。変更前の小さなCPU入力コピーをケース当たり1つだけ保持する修正を追加し、複数ケースの参照解放・選択plan・元の履歴でのcost再計算を回帰試験しました。修正後の公式10ケースは標準予算で10/10成功、exit 0、status succeeded、全ケースでagent移動、GPU peak allocated 471918592 bytes。10本のMP4を実デコードし各22〜25種類のフレームを確認。ユーザー指示で学習は停止済み。CPU全体119合格、CUDA専用5件も別途実行して合格。BT途中checkpointの同10ケース確認を継続中です。50ケース完走やRTX側のBT重みの確認とは主張しません。

**Comment by sige0002** [comment]
修正を5f19742でmainへpushしました。最終コードでGPU有効の全回帰126件合格・スキップ0件（警告236件は記録）。実機は固定confirm先頭10件・標準CEM予算で公式10/10、ローカル旧BT 15k 4/10、新Raw 10k 5/10、ランダム0/10。全実行のexit 0・status succeeded・結果JSON、開始/Goal hash一致、各ケースのagent移動を確認。各10本のMP4もデコードしてフレーム変化を確認しました。Rawは修正前1ケース確認と同じ10k重みです。データ全量hashは再計算していません。失敗記録と詳細は mylewm/docs/reports/PUSHT_ISSUE20.ja.md。学習はユーザー指示で停止済み（ログ74,504更新、最後の保存70,000更新）、評価プロセスも終了済みです。RTX側の該当BT 10k重み、および修正後50ケース完走は未確認なのでIssueは開いたままにします。これら10ケースの動作確認を方式の性能比較とは扱いません。

**Comment by sige0002** [comment]
追試を完了し、a46535fで実測記録をpushしました。旧6272e47を無改変worktreeで評価し、同一公式重み・同じ固定50ケース・seed42・依存環境・標準CEM予算で旧0/50→現行5f19742 45/50（90%）。開始/Goal画像の実HDF5との一致も各0/50→50/50。動画各50本をデコードし、旧版のほぼ静止した保存と修正後のフレーム変化を確認。低成功率と画像/動画経路の不具合を旧コミットで再現できました。現行BT100kは44/50（88%）、両方exit0/status succeeded・全ケースでagent移動。比較CLIの条件/ソース/画像hash検査も通過しました。ただし実物理初期状態を追加照合すると13ケースのobject位置/角度に微差（最大x1.1803、y0.8509、角度0.01486rad）があり、既存比較CLIはこの差を検査していません。この13件の成否は両者同じですが、影響なしとは証明できません。原因は未確定のため厳密初期状態の一致確認を残し、Issueは開いたままにします。詳細と全出力パスは mylewm/docs/reports/PUSHT_ISSUE20.ja.md。今回の追試では評価コードを追加変更していません。

**Comment by sige0002** [comment]
元の評価画像・動画経路の不具合は5f19742で修正し、同一公式重み・固定50ケースの旧0/50→修正後45/50の追試を記録済みです。ユーザーの指示に基づき、この不具合のIssueとしてクローズします。今回の追加コード変更・追加評価はありません。物理初期状態の微差の解消を意味するクローズではなく、厳密な方式比較の制約は残ります（保存済み14万更新の実step前状態にも差を確認）。過去の生記録は保持します。今後の作業はSceneの#21〜#23を優先します。
